src/System.Net.Primitives/src/System/Net/Sockets/SocketError.cs

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Net;
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.Net.Sockets
+{
+    public enum SocketError : int
+    {
+        Success = 0,
```

Comment:
@pgavlin I would prefer if the error codes are added in the winsock interop and this implementation references them. It makes a big difference in debugging issues. The error codes were present in the original file as comments such as:

`        Interrupted = (10000 + 4),      //WSAEINTR`

We could either keep the comments in or have something like:
`        Interrupted = Interop.winsock.WSAEINTR;`

Unfortunately, given that the codes don't match the NIX errno.h ones we'll need an adapter for xplat.